src/traces/image/attributes.js

```diff
@@ -51,6 +51,17 @@ module.exports = extendFlat({
             'otherwise it defaults to `rgb`.'
         ].join(' ')
     },
+    zsmooth: {
+        valType: 'enumerated',
+        values: ['fast', false],
+        dflt: false,
+        role: 'info',
+        editType: 'plot',
+        description: [ 
+            'Picks a smoothing algorithm used to smooth `z` data.',
+            'This only applies for image traces that use the `source` attribute.'
+        ].join(' ') 
```

Comment:
The heapmap trace has `['fast', 'best', false]`. I'm not sure what `false` means there, but would it not make more sense to have 'fast' and 'best' here, which would resolve to pixelated and "auto", respectively?

We should also add a note that the result may be browser dependent.

Reply:
Ha, we really need better documentation of those options for heatmaps, but here's what they mean:
- `false` means no smoothing, each data point is a rectangle (or "brick") of constant color.
- `'fast'` means we let the browser interpolate smoothly between one data value and the next (so yes, the result may be browser-dependent). This only works when all pixels are the same size, and it can cause problems if two neighboring points have very different value, as the browser will probably interpolate linearly in RGB space and intermediate values may not be present in the colorscale at all.
- `'best'` means we smooth by calculating the color at each screen pixel independently, first with bilinear interpolation of data values, then we map the result onto the colorscale. This can be slow, but it handles nonuniform x or y spacing and ensures every interpolated value is in the colorscale.

So what you've implemented here - assuming the browser behaves as expected - corresponds to `false` (pixelated) and `'fast'` (smoothed). `'best'` probably doesn't make sense for images unless there are important browsers that we can't get to behave correctly with `'fast'`. But we don't support nonuniform pixels in image traces, and there's no colorscale so the interpolation we would do manually is likely the same as the browser does, unless perhaps we wanted to support interpolating in HSL space.